modules/store/src/action_creator.ts

```diff
@@ -124,7 +124,14 @@ export function createAction<T extends string, C extends Creator>(
   }
 }
 
-export function props<P extends object>(): ActionCreatorProps<P> {
+export function props<
+  P extends R extends any
+    ? P extends object
+      ? NotAllowedCheck<P>
+      : unknown
+    : unknown,
+  R = any
+>(): ActionCreatorProps<P> {
```

Comment:
Hey @david-shortman , nice idea 🙂

Try this: [playground Link](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAYwgOwM7wIZShgnqgQSmADkIYCAbSiAd2ABMBZVAczgF44ByDBGAJYpExDDGhwwUCGFSIMyZOTgAjYHAUbsebgG4AUDFxh1RHPiKlyVGvQac4RkxABmW84WJkK1OoxasBvqgkLCIKOiOxsAACtImsLgAkqjeNn7MbA68-ELIIsBiElIycggKSvAAFhgAbuoYkvHAiXDIGAC2jHAABk7APXqG0XAAKtFxMi1GKWm+dg79rlEmkwkzqdbz-mxBIdDwSGjwwB1gRgDyKgBWwPyeVj62O+xcOYLCCKLiUE2l8oplGoNPlTudcHAIDc7jAhv04ABRM6XaH3SxzZ72LhLNxglG3NFeLaYgJ7cAHcLHJoCDoCQT1B4YjIBbJ8D75L6FH5-WQAyqqBrU2n09S1DCUACuwDhIziNLpAgZ6OJGUW0WWUnlIsZKrspP0wxMcDlwsV6i46CgAmQ7AAPm0JR01L97SoBKxrfBXRAIJRCvl7cgJdQ4PaJcgGMAXNbGGTQvB4Uy7ABhKp3ADWAB5RnBQDBgBG5FCCTAAHwOHN5gsMOQKXAAbQAuvo4HAAPxwMx4HVPDItuAALjGuZA+cLcAA3itgEO63AAL79jvjVbNRKzXWMftD9PAXDLSuj6tyZDAepQJeI5G4K4lnvpOzbuDh9NKWjIOMUxObhhrVCpjNs3LDh+0PMca0hVEYH7VsOyTRgAIQLNRlLGDW0HYcq3HE0FXqND0PbY0rVNJUiV7R8CIIocXzfD8DX2MIXHDXJhBKWRM37P8R3AuQ-0Q9MABpOPif80yQhx4N-ET+MzP9UNLAAKABKSd+2IGAJSgfIJ3nDRa2QXADEXfQ2NQTMp36IdLWtQI4BAIcgydFoF0UpSglMzNrJtJtXPckTzPnXyDQ8xznV8oA) 

```suggestion
export function props<
  Props extends PropsCheck,
  PropsCheck = NotAllowedPropsCheck<Props>
>(): ActionCreatorProps<Props> {
```